app/models/house.rb

```diff
@@ -3,10 +3,27 @@ class House < ApplicationRecord
   has_many :home_owners_associations, through: :hood
   has_one :house_metadatum, dependent: :destroy
 
+  delegate :zpid, to: :house_metadatum, allow_nil: true
+
+  after_initialize :set_price_history
+
   PRICE_HISTORY_DATE_FORMAT = "%Y-%m-%d".freeze
 
   def valuation_on(date)
     normalized_date = date.strftime(PRICE_HISTORY_DATE_FORMAT)
     price_history.dig(normalized_date).to_f
   end
+
+  def add_valuation(date, price)
+    normalized_valuation_date = date.strftime(PRICE_HISTORY_DATE_FORMAT)
+    price_history[normalized_valuation_date] = price.to_f
```

Comment:
Don't we need to save the record after doing this?

Reply:
I thought about it at first, but then figured this could be like assigning a variable -- it doesn't necessarily have to save every time you add a valuation.

Reply:
So the caller is responsible for calling `.save!` ?

Reply:
Yup!